hooks/secondary_publish_tk-houdini.py

```diff
@@ -116,3 +146,130 @@ def execute(
+            "published_file_type": tank_type
+        }
+        tank.util.register_publish(**args)
+    
+    def __publish_rendered_images(self, item, output, work_template, primary_publish_path, 
+                                  sg_task, comment, thumbnail_path, progress_cb):
+        """
+        Publish rendered images and register with Shotgun.
+        
+        :param item:                    The item to publish
+        :param output:                  The output definition to publish with
+        :param work_template:           The work template for the current scene
+        :param primary_publish_path:    The path to the primary published file
+        :param sg_task:                 The Shotgun task we are publishing for
+        :param comment:                 The publish comment/description
+        :param thumbnail_path:          The path to the publish thumbnail
+        :param progress_cb:             A callback that can be used to report progress
+        """
+
+        # determine the publish info to use
+        #
+        progress_cb(10, "Determining publish details")
+
+        # get the current scene path and extract fields from it
+        # using the work template:
+        scene_name = str(hou.hipFile.name())
+        scene_path = os.path.abspath(scene_name)
+        fields = work_template.get_fields(scene_path)
+        publish_version = fields["version"]
+        tank_type = output["tank_type"]
+
+        # this is pretty straight forward since the publish file(s) have
+        # already been created (rendered). We're really just populating the
+        # arguments to send to the sg publish file registration below.
+        publish_name = item["name"]
+
+        # we already determined the path in the scan_scene code. so just 
+        # pull it from that dictionary.
+        other_params = item["other_params"]
+        paths = other_params["paths"]
+        node = other_params["node"]
+
+        if not paths:
+            raise TankError(
+                "No rendered images found for node '%s'." % (node.path(),)
+            )   
+        elif len(paths) > 1:
+            raise TankError(
+                "Found multiple potential rendered image paths for node '%s'." 
+                "Skipping these paths:\n  '%s'" %
+                (node.path(), "\n  ".join(paths))
+            )
+
+        publish_path = paths[0]
+
+        # register the publish:
+        progress_cb(75, "Registering the publish")        
+        args = {
+            "tk": self.parent.tank,
+            "context": self.parent.context,
+            "comment": comment,
+            "path": publish_path,
+            "name": publish_name,
+            "version_number": publish_version,
+            "thumbnail_path": thumbnail_path,
+            "task": sg_task,
+            "dependency_paths": [primary_publish_path],
+            "published_file_type": tank_type
+        }
+        tank.util.register_publish(**args)
```

Comment:
Yes, I think so. But the differences are right in the middle of the methods. I think this is probably true for a lot of publish hooks across the various DCCs. I'm inclined to leave it unless you feel strongly that it should be cleaned up.